sdk/test-utils/recorder/ASSET_SYNC_MIGRATION.md

```diff
@@ -0,0 +1,73 @@
+# Migrating recordings to the `azure-sdk-assets` repository
+
+## Some background
+
+Recordings take up a large amount of space in our repository and generate a lot of churn. The asset sync project, owned by our Engineering System team, is a solution to this problem. Leveraging and extending the existing test proxy, the asset sync extension adds support for moving the recordings outside of the main azure-sdk-for-js repo.
+
+## Performing the migration
+
+The package you are migrating needs to be using the new version of the recorder that uses the test proxy (`@azure-tools/test-recorder@^2.0.0`). Most packages are using the new recorder already; if yours is not you should migrate as soon as possible. More detail on migrating to the new recorder can be found in the [recorder 2.0 migration guide].
+
+To migrate, there is just one command to run. With [`dev-tool`] and [Powershell] installed, run the following in your package directory:
+
+```bash
+$ npx dev-tool test-proxy migrate --initial-push
+```
+
+Once this is done, validate that your recorded tests still pass, and create a PR with the changes. That's it!
+
+## Workflow with asset sync enabled
+
+With asset sync enabled, there is one extra step that must be taken before you create a PR with changes to recorded tests: you must push the new recordings to the assets repo. This is done with the following command:
+
+```bash
+$ npx dev-tool test-proxy push
+```
+
+This command will:
+
+1. Push your local recordings to a tag in the `Azure/azure-sdk-assets` repo, and
+1. Update the `assets.json` in your package root to reference the newly created tag.
+
+You should stage and commit the `assets.json` update as part of your PR. If you don't run the `push` command before creating a PR, the CI (and anyone else who tries to run your recorded tests) will use the old recordings, which will cause failures.
+
+This diagram describes the new workflow (new steps highlighted):
+
+```mermaid
+graph LR
```

Comment:
Beautiful.

Isn't TD better for viewing?

Reply:
Yeah good point, the LR one is kind of long...